tests/AutoMapperTest.php

```diff
@@ -475,6 +475,43 @@ public function testConstructor(): void
         self::assertTrue($userDto->getConstructor());
     }
 
+    public function testConstructorAndRelationMissing(): void
+    {
+        $user = ['name' => 'foo'];
+        $this->expectException(MissingConstructorArgumentsException::class);
+
+        /** @var Fixtures\UserConstructorDTOWithRelation $userDto */
+        $userDto = $this->autoMapper->map($user, Fixtures\UserConstructorDTOWithRelation::class);
+    }
+
+    public function testConstructorAndRelationMissing2(): void
+    {
+        $user = ['name' => 'foo', 'int' => ['foo' => 1]];
```

Comment:
I changed the input array, passing an Object will not work as we expecte pure base type when mapping from array, if there is something else then a transformer should be provided